// extra grpc services for testing purposes

syntax = "proto3";
package io.bisq.protobuffer;
import "pb.proto";
option java_package = "bisq.proto.grpc";
option java_multiple_files = true;

/*
* The DevCommmands service is provided for development only
*/
service DevCommands {
    rpc SendProto (SendProtoRequest) returns (SendProtoReply) {
    }
}

message SendProtoRequest {
    NodeAddress destination_node_address = 1;
    NetworkEnvelope network_envelope = 2;
}

message SendProtoReply {
    bool success = 1;
    string error_message = 2;
}

service UserManagerCommands {
    rpc SwitchUser (SwitchUserRequest) returns (SwitchUserReply) {
    }

    rpc CreateNewUser (CreateNewUserRequest) returns (CreateNewUserReply) {
    }

    rpc DeleteUser (DeleteUserRequest) returns (DeleteUserReply) {
    }

    rpc SetUserAlias (SetUserAliasRequest) returns (SetUserAliasReply) {
    }

    rpc GetUsersList (CreateNewUserRequest) returns (GetUsersListReply) {
    }

    rpc RestoreUser (RestoreUserRequest) returns (RestoreUserReply) {
    }
}

message SwitchUserRequest {
    string user_id = 1;
}

message SwitchUserReply {
}

message CreateNewUserRequest {
}

message CreateNewUserReply {
    string user_id = 1;
}

message DeleteUserRequest {
    string user_id = 1;
    bool delete_data = 2;
}

message DeleteUserReply {
    string new_user_id = 1;
    bool created_new_user = 2;
}

message SetUserAliasRequest {
    string user_id = 1;
    string alias = 2;
}

message SetUserAliasReply {
}

message GetUsersListRequest {
}

message GetUsersListReply {
    repeated BriefUserInfo users_list = 1;
}

message BriefUserInfo {
    string user_id = 1;
    string alias = 2;
}

message RestoreUserRequest {
    string user_id = 1;
}

message RestoreUserReply {
}
